### DISCUSSION TITLE
Can't use dependency injection with service

### DISCUSSION THREAD

**Discussion by angelagastal** [CoreWCF/CoreWCF#458] [Q&A]
Does anyone have an example of how to accomplish this?

I tried suggestions made on https://github.com/CoreWCF/CoreWCF/issues/56 but no luck.

I'm using the latest CoreWCF 0.3.0 and .NET 5.0

**Comment by mconnew** [comment]
Can you provide the code that you have tried to do so far? You should just be able to do `services.AddSingleton<ServiceType>();` in your ConfigureServices method and then in the UseServiceModel delegate, you call `serviceBuilder.AddService<ServiceType>();` and it should just pick it up. You can also provide a pre-created instance or use a factory method or any of the other instantiation mechanisms available to DI.

**Comment by dioptryk** [comment]
@angelagastal please be careful to use the same type of generic arguments in both `IServiceCollection.AddSingleton` and `IServiceBuilder.AddService` (rather than passing both the interface and implementation in one of them). I had the same issue and this fixed it, my service has all necessary stuff injected like it should.

**Reply by alexvazquez** [reply]
> e the same type of generic arguments in both `IServiceCollection.AddSingleton` and `IServiceBuilder.AddService` (rather than passing both the interface and implementation in one of the

Did you find a way? I need to access IConfiguration inside one of my services.

**Reply by mconnew** [reply]
This works. What he means here is don't do `IServiceCollection.AddSingleton<IService,Service>()` (or some variation), but use `IServiceCollection.AddSingleton<Service>()` as CoreWCF asks DI for the type of your service implementation and NOT for the type of the contract interface.

**Reply by alexvazquez** [reply]
Appreciate it. Working right now :)

**Comment by Danielku15** [comment]
you might be interested into #467 I entered and #468 I opened to address it. 😉
